jkind/src/jkind/solvers/yices2/Yices2Solver.java

```diff
@@ -42,8 +42,8 @@ protected String getSolverExecutable() {
 	@Override
 	public void initialize() {
 		send("(set-option :produce-models true)");
-		send("(set-option :produce-unsat-cores true)");
 		if (linear) {
+			send("(set-option :produce-unsat-cores true)");
```

Comment:
Do you need to update the solver so it doesn't ask for an unsat core? And you need to handle the return values in that case.